packages/components/src/button/index.native.js

```diff
@@ -91,108 +52,130 @@ export function Button( props ) {
 		label,
 		shortcut,
 		tooltipPosition,
+		styles = {},
 	} = props;
 
-	const isDisabled = ariaDisabled || disabled;
+	const colorScheme = usePreferredColorScheme();
 
-	const buttonViewStyle = {
-		opacity: isDisabled ? 0.3 : 1,
-		...( fixedRatio && styles.fixedRatio ),
-		...( isPressed ? styles.buttonActive : styles.buttonInactive ),
-	};
+	const isDisabled = ariaDisabled || disabled;
 
-	const states = [];
-	if ( isPressed ) {
-		states.push( 'selected' );
-	}
+	const themedStyle = usePreferredColorSchemeStyleBem( {
+		...baseStyles,
+		...styles,
+	} );
 
-	if ( isDisabled ) {
-		states.push( 'disabled' );
-	}
+	/* eslint-disable dot-notation */
+	const containerStyle = themedStyle[ 'button__container' ];
+
+	const buttonStyle = useMemo(
+		() => ( {
+			opacity: isDisabled ? 0.3 : 1,
+			...( fixedRatio && { aspectRatio: 1 } ),
+			...( isPressed
+				? themedStyle[ 'button__button--active' ]
+				: themedStyle[ 'button__button--inactive' ] ),
+		} ),
+		[ fixedRatio, isDisabled, isPressed ]
+	);
 
-	const subscriptInactive = getStylesFromColorScheme(
-		styles.subscriptInactive,
-		styles.subscriptInactiveDark
+	const subscriptStyle = useMemo(
+		() =>
+			isPressed
+				? themedStyle[ 'button__subscript--active' ]
+				: themedStyle[ 'button__subscript--inactive' ],
+		[ isPressed ]
 	);
 
+	const iconColor = themedStyle[ 'button__icon' ]?.color;
+	/* eslint-enable dot-notation */
+
 	const newChildren = Children.map( children, ( child ) => {
 		return child
 			? cloneElement( child, {
-					colorScheme: props.preferredColorScheme,
+					colorScheme,
 					isPressed,
 			  } )
 			: child;
 	} );
 
-	// Should show the tooltip if...
```

Comment:
TBH this logic was tricky to follow. Here is what I deduced in terms of examples:
```
// Assume isDisabled = false in the following examples, 
// if isDisable _is_ true then the button will not use Tooltip.

// These *will* use a Tooltip
<Button shortcut='foobar' /> // shortcut is specified , other props ignored
<Button label='foobar' />  // has label and showTooltip is not explicitly false 
<Button label='foobar' showTooltip={true} ><Text>child</Text> // has children but showTooltip is explicitly true 

// These *will not* use a Tooltip
<Button /> // no label or shortcut 
<Button label='foobar' showTooltip={false} /> // showTooltip is explicitly false 
<Button label='foobar'><Text>child</Text></Button> // has children, showTooltip is not specified 
```